weave-js/src/common/util/markdown.ts

```diff
@@ -16,20 +17,41 @@ import visit from 'unist-util-visit';
 
 import {blankifyLinks, shiftHeadings} from './html';
 
-const sanitizeRules = _.merge(gh, {
+// exported only for tests
+export const DEFAULT_SANITIZATION_SCHEMA = _.merge(gh, {
   attributes: {'*': ['className', 'style']},
 });
 
-// Hackily convert markdown to text
-export function markdownToText(markdown: string) {
-  const html = generateHTML(markdown);
+const SANITIZATION_SCHEMAS_FOR_RULES: Record<keyof SanitizationRules, Schema> =
```

Comment:
the logic in this file changed around how we know when to apply the allowedScopeStyles, and how it's applied. 

* I wanted sanitization rules to allow arbitrary json since sanitization rules can be a bit complex. 
* I simplified the build schema method since the previous method was very clever, but it was hard to understand what was really happening. Since this is security related code, I preferred strongly perferred simplicity for the code here.